File: app/src/main/java/com/example/rpanaquecavana/gitandroid/ListAdapter.java
package com.example.rpanaquecavana.gitandroid;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.rpanaquecavana.gitandroid.Modelos.Item;
import com.example.rpanaquecavana.gitandroid.Modelos.RepoGit;

import java.util.ArrayList;
import java.util.List;

public class ListAdapter extends RecyclerView.Adapter<ListAdapter.ViewHolderDatos>
{
    private Context context;
    private ArrayList<Item> repodata;

    public ListAdapter(ArrayList<Item> repodata) {
        this.repodata = repodata;
    }

    @NonNull
    @Override
    public ViewHolderDatos onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.detalle,parent,false);
        ViewHolderDatos viewHolderDatos = new ViewHolderDatos(view);
        return viewHolderDatos;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderDatos holder, int position) {
       // RepoGit repoGit = repodata.get(position);
        holder.txtuser.setText(repodata.get(position).getOwner().getLogin());
        holder.txtdescripcion.setText(repodata.get(position).getDescription());
        holder.txttitulo.setText(repodata.get(position).getName());

    }

    @Override
    public int getItemCount() {
        return repodata.size();
    }

    public class ViewHolderDatos extends RecyclerView.ViewHolder
    {
        TextView txttitulo, txtdescripcion, txtuser;
        ImageView imagen;

        public ViewHolderDatos(@NonNull View itemView) {
            super(itemView);

            txttitulo = itemView.findViewById(R.id.txttitle);
            txtdescripcion = itemView.findViewById(R.id.txtdescription);
            txtuser = itemView.findViewById(R.id.txtusername);
            imagen = itemView.findViewById(R.id.icono);
        }

        public void asignardatos(String repoGit)
        {
            txtuser.setText(repoGit);
        }
    }
}

File: app/src/main/java/com/silvioapps/githubkotlin/features/list/adapters/ListAdapter.kt
package com.silvioapps.githubkotlin.features.list.adapters

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.silvioapps.githubkotlin.BR
import com.silvioapps.githubkotlin.R
import com.silvioapps.githubkotlin.features.list.models.ListModel
import com.silvioapps.githubkotlin.features.shared.listeners.ViewClickListener
import com.silvioapps.githubkotlin.features.shared.utils.Utils

class ListAdapter (list_ : MutableList<ListModel>, viewClickListener_ : ViewClickListener) : RecyclerView.Adapter<ListAdapter.BindingViewHolder>() {
    private var list: MutableList<ListModel>

    companion object{
        private lateinit var viewClickListener: ViewClickListener
    }

    init{
        this.list = list_
        viewClickListener = viewClickListener_
    }

    class BindingViewHolder(view : View) : RecyclerView.ViewHolder(view){
        var viewDataBinding : ViewDataBinding? = null

        init{
            viewDataBinding = DataBindingUtil.bind<ViewDataBinding>(view)
            Utils.setClickListeners(view, viewClickListener, listOf<ListModel>())
        }
    }

    override fun onCreateViewHolder(parent : ViewGroup, viewType: Int) : BindingViewHolder{
        val view = LayoutInflater.from(parent.context).inflate(R.layout.main_list_layout, parent, false)
        return BindingViewHolder(view)
    }

    override fun onBindViewHolder(holder : BindingViewHolder, position : Int) {
        Utils.setTags(position, holder.itemView)

        if(list.size > position) {
            val listModel : ListModel = list.get(position)
            holder.viewDataBinding?.setVariable(BR.listModel, listModel)
            holder.viewDataBinding?.executePendingBindings()
        }
    }

    override fun getItemCount() : Int{
        return list.size
    }
}
